src/nvim/tui/input.c

```diff
@@ -357,15 +358,17 @@ static bool handle_forced_escape(TermInput *input)
 static void set_bg_deferred(void **argv)
 {
   char *bgvalue = argv[0];
-  if (starting) {
-    // Wait until after startup, so OptionSet is triggered.
-    loop_schedule(&main_loop, event_create(set_bg_deferred, 1, bgvalue));
-    return;
-  }
   if (!option_was_set("bg") && !strequal((char *)p_bg, bgvalue)) {
     // Value differs, apply it.
-    set_option_value("bg", 0L, bgvalue, 0);
-    reset_option_was_set("bg");
+    if (starting) {
+      // Wait until after startup, so OptionSet is triggered.
```

Comment:
> // Wait until after startup, so OptionSet is triggered.

nitpicking, but it is more like *"schedule the bg update, so OptionSet is triggered"* now

Reply:
shorter is better if it does not confuse.

Reply:
hm, but we do not *wait* for something, we are scheduling relying on the fact that all autocommands run after startup. It depends on the reader of the code though, but someone could be confused.

Reply:
The code is clear , though. Comment shouldn't merely restate what the code already says.

Reply:
> Comment shouldn't merely restate what the code already says.

it does not.
 it says that we do if the editor is not started yet and why.
```c
if(starting){
// defer the update, so Option is triggered`
...
```
and the comment is needed because in the next line we do not have some specific API for schedule the things but have `do_cmdline_cmd` what rarely have being used for this purpose(deferring the things).

but I got your point